packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/FlatTreeNode.ts

```diff
@@ -49,3 +59,41 @@ function getFlatNodes(nodes: PresentationTreeNode[], level: number) {
   });
   return flatNodes;
 }
+
+export function getErrors(rootNodes: PresentationTreeNode[]): ErrorNode[] {
+  return rootNodes.flatMap((rootNode) => {
+    if (!isPresentationHierarchyNode(rootNode)) {
+      return [{ parent: undefined, error: rootNode, expandTo: (expandNode) => expandTo(expandNode, []) }];
+    }
+    if (rootNode.children === true) {
+      return [];
+    }
+    return getErrorNodes(rootNode, !rootNode.isExpanded ? [rootNode.id] : []);
+  });
+}
+
+function getErrorNodes(parent: PresentationHierarchyNode, path: string[]) {
+  const errorList: ErrorNode[] = [];
+
+  if (parent.children === true) {
+    return [];
+  }
+
+  parent.children.forEach((node) => {
+    if (!isPresentationHierarchyNode(node)) {
+      errorList.push({ parent, error: node, expandTo: (expandNode) => expandTo(expandNode, path) });
+      return;
+    }
+
+    if (node.children !== true) {
+      const childErrorList = getErrorNodes(node, [...path, ...(!parent.isExpanded ? [parent.id] : [])]);
```

Comment:
You can avoid copying array multiple times for each child node with error.
```suggestion
  const pathToChild = [...path, ...(!parent.isExpanded ? [parent.id] : [])];
  parent.children.forEach((node) => {
    if (!isPresentationHierarchyNode(node)) {
      errorList.push({ parent, error: node, expandTo: (expandNode) => expandTo(expandNode, path) });
      return;
    }

    if (node.children !== true) {
      const childErrorList = getErrorNodes(node, pathToChild);
```